content/guides/advanced/testing-with-jest-and-bankrun.md

```diff
@@ -0,0 +1,380 @@
+
+## Anchor test
+
+What `anchor test` does is that Anchor will perform the command defined in
+`anchor.toml` file.
+
+```bash
+test = "yarn run ts-mocha -p ./tsconfig.json -t 1000000 tests/**/*.ts"
+```
+
+This will run the tests in the tests directory using the ts-mocha command with a
+timeout of 1,000,000 milliseconds.
+
+What anchor does here is that it starts up a local validator, deploys the
+program from your anchor workspace and runs the test against the defined network
+in the `anchor.toml`.
+
+> Tip: You can also run `anchor test --detach`. This will let the validator
+> continue running after the tests have finished which lets you inspect your
+> transactions in the
+> [solana explorer](https://explorer.solana.com/?cluster=custom).
+
+You can also define your own test command in the `anchor.toml` file. For example
+you can first run the mocha tests against the local validator and then run the
+jest tests using bankrun by combining them using `&&`:
+
+```bash
+test = "yarn run ts-mocha -p ./tsconfig.json -t 1000000 tests/**/*.ts && yarn run jest"
+```
+
+This would run the mocha tests first and then run the jest tests.
+
+You can also define your own test command. For example:
+
+```bash
+super_test = "yarn run ts-mocha -p ./tsconfig.json -t 1000000 only_super_test.ts"
+jest_tests = "yarn test"
+```
+
+These you would then run using:
+
+```bash
+anchor run super_test
+anchor run jest_tests
+```
+
+> Note though that in this case anchor does not start a local validator for you.
+> So you will need to deploy the program yourself to your local cluster or run
+> them against a public test network. Also Anchor environment variables will
+> only be available when run trough the `anchor test` commands not when running
+> tests with `yarn test` for example.
+
+## Migrating from Mocha to Jest
+
+In this part we will learn how to migrate from [Mocha](https://mochajs.org/) to
+[Jest](https://jestjs.io/). Jest is another Javascript testing framework similar
+to Mocha. It has already and integrated test runner so you don't need to use
+Chai anymore.
+
+First you need to install Jest:
+
+```bash
+yarn add --dev jest
+```
+
+Then you add a new command to your `package.json`:
+
+```json
+{
+  "scripts": {
+    "test": "jest"
+  }
+}
+```
+
+Then you can run the tests with:
+
+```bash
+yarn test
+```
+
+Since we want to run our tests with Typescript we need to install the `ts-jest`
+package and also create a Jest configuration file:
+
+```bash
+yarn add --dev ts-jest @jest/globals @types/jest
+yarn ts-jest config:init
+```
+
+This will create a `jest.config.js` file in your project. Now you can update
+your `anchor.toml` file to run the Jest tests:
+
+```bash
+test = "yarn test"
+```
+
+### Jest Troubleshooting
+
+1. In case your get a
+   `SyntaxError: Cannot use import statement outside a module` error you either
+   did not create a jest config or you need to add the following to your
+   `jest.config.js` file:
```

Comment:
```suggestion
1. In case you get a `SyntaxError: Cannot use import statement outside a module`
   error, you either did not create a jest config or you need to add the
   following to your `jest.config.js` file:
```